builder/cloudstack/step_create_instance.go

```diff
@@ -30,13 +30,14 @@ func (s *stepCreateInstance) Run(state multistep.StateBag) multistep.StepAction
 	client := state.Get("client").(*cloudstack.CloudStackClient)
 	config := state.Get("config").(*Config)
 	ui := state.Get("ui").(packer.Ui)
+	source := state.Get("source").(string)
```

Comment:
Since this value is used only once, I would just make this `state.Get` call inline were we actually need it.